pages/resources/[page].js

```diff
@@ -1,44 +1,55 @@
-import PropTypes from 'prop-types';
+import { useEffect, useState } from 'react';
+import { useRouter } from 'next/router';
+import get from 'lodash/get';
+import isNumber from 'lodash/isNumber';
 import Content from 'components/Content/Content';
 import Head from 'components/head';
 import HeroBanner from 'components/HeroBanner/HeroBanner';
-import ResourceCard from 'components/Cards/ResourceCard/ResourceCard';
 import Pagination from 'components/Pagination/Pagination';
+import ResourceCard from 'components/Cards/ResourceCard/ResourceCard';
 import { getResourcesPromise } from 'common/constants/api';
 import styles from '../styles/resources.module.css';
 
-ResourcesPage.propTypes = {
-  currentPage: PropTypes.number.isRequired,
-  pathname: PropTypes.string.isRequired,
-  resources: PropTypes.arrayOf(
-    PropTypes.shape({
-      id: PropTypes.number, // integer, unique ID
-      name: PropTypes.title,
-      notes: PropTypes.string, // possibly null
-      url: PropTypes.string.isRequired,
-      upvotes: PropTypes.number,
-      downvotes: PropTypes.number,
-    }),
-  ).isRequired,
-  totalPages: PropTypes.number.isRequired,
-};
+function ResourcesPage() {
+  const router = useRouter();
+  const { pathname, query } = router;
+  const currentPage = parseInt(query.page, 10);
+
+  if (!isNumber(currentPage)) {
+    // TODO: Handle situation where user tried some funny business, like `/resources/fuck-you`
+    throw new Error(`'currentPage' is not a number`);
+  }
 
-ResourcesPage.getInitialProps = async ({ pathname, query }) => {
-  const { page = 1 } = query;
+  const [isLoading, setIsLoading] = useState(true);
+  const [resources, setResources] = useState([]);
+  const [totalPages, setTotalPages] = useState(currentPage);
 
-  const response = await getResourcesPromise({ page });
+  useEffect(() => {
+    getResourcesPromise({ page: query.page })
+      .then(response => {
+        const fetchedResources = get(response, 'data.data', []);
```

Comment:
get the response.data.data or else return an empty array?

Reply:
Aye.